pkg/vm/createCustom.go

```diff
@@ -55,3 +78,104 @@ func loadCustomGenesis(app *application.Avalanche, genesisPath string) ([]byte,
 	genesisBytes, err := os.ReadFile(genesisPath)
 	return genesisBytes, err
 }
+
+func SetCustomVMSourceCodeFields(app *application.Avalanche, sc *models.Sidecar, customVMRepoURL string, customVMBranch string, customVMBuildScript string) error {
+	var err error
+	if customVMRepoURL != "" {
+		ux.Logger.PrintToUser("Checking source code repository URL %s", customVMRepoURL)
+		if err := prompts.ValidateURL(customVMRepoURL); err != nil {
+			ux.Logger.PrintToUser("Invalid repository url %s: %s", customVMRepoURL, err)
+			customVMRepoURL = ""
+		}
+	}
+	if customVMRepoURL == "" {
+		customVMRepoURL, err = app.Prompt.CaptureURL("Source code repository URL")
+		if err != nil {
+			return err
+		}
+	}
+	if customVMBranch != "" {
+		ux.Logger.PrintToUser("Checking branch %s", customVMBranch)
+		if err := prompts.ValidateRepoBranch(customVMRepoURL, customVMBranch); err != nil {
+			ux.Logger.PrintToUser("Invalid repository branch %s: %s", customVMBranch, err)
+			customVMBranch = ""
+		}
+	}
+	if customVMBranch == "" {
+		customVMBranch, err = app.Prompt.CaptureRepoBranch("Branch", customVMRepoURL)
+		if err != nil {
+			return err
+		}
+	}
+	if customVMBuildScript != "" {
+		ux.Logger.PrintToUser("Checking build script %s", customVMBuildScript)
+		if err := prompts.ValidateRepoFile(customVMRepoURL, customVMBranch, customVMBuildScript); err != nil {
+			ux.Logger.PrintToUser("Invalid repository build script %s: %s", customVMBuildScript, err)
+			customVMBuildScript = ""
+		}
+	}
+	if customVMBuildScript == "" {
+		customVMBuildScript, err = app.Prompt.CaptureRepoFile("Build script", customVMRepoURL, customVMBranch)
+		if err != nil {
+			return err
+		}
+	}
+	sc.CustomVMRepoURL = customVMRepoURL
+	sc.CustomVMBranch = customVMBranch
+	sc.CustomVMBuildScript = customVMBuildScript
+	return nil
+}
+
+func checkGitIsInstalled() error {
+	if err := exec.Command("git").Run(); errors.Is(err, exec.ErrNotFound) {
```

Comment:
nit: run `git version` not `git`

Reply:
no needed, it is searching for ErrNotFound, it is failing is binary is not found

Reply:
why do you prefer git version ?